File: 2017/s2.py
#!/bin/python

recordings = int(input(), 10)  # Number of recordings
tides = list(map(int, input().split()))  # List of given recordings

tides.sort()  # Recordings are ordered generally

# Calculate the positions to slice
slice = (len(tides) // 2) if recordings % 2 == 0 else len(tides) // 2 + 1

low, high = (  # Low tides are on the first half, high tides on the second half
    tides[0:slice],
    tides[slice:],
)

low.reverse()  # Low tides go from greatest to least

for i in range(len(high)):  # Print each tide accordingly
    print(low[i], high[i], end=" ")

if recordings % 2 == 1:  # If an extra low tides exists at the end, print it
    print(low[-1])

File: 2017/s2.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::vector;

int main() {
    uint recordings;    // Number of recordings
    vector<uint> tides; // List of given recordings

    cin >> recordings;

    for (uint i = 0; i < recordings; i++) {
        uint tide;
        cin >> tide;

        tides.push_back(tide);
    }

    std::sort(tides.begin(), tides.end()); // Recordings are ordered generally
    uint slice = recordings % 2 == 0 ? tides.size() / 2
                                     : tides.size() / 2 + 1; // Calculate the positions to slice

    auto low =
        vector<uint>(tides.begin(), tides.begin() + slice); // Low tides are on the first half
    auto high = vector<uint>(tides.begin() + slice, tides.end()); // High tides on the second half

    std::reverse(low.begin(), low.end()); // Low tides go from greatest to least

    for (uint i = 0; i < high.size(); i++) { // Print each tide accordingly
        cout << low[i] << " " << high[i] << " ";
    }

    if (recordings % 2 == 1) { // If an extra low tides exists at the end, print it
        cout << low.back();
    }

    return 0;
}
